### ISSUE TITLE
Prediction interval with XGBoost

### ISSUE THREAD

**Issue by Shafi2016** [AdrianAntico/RemixAutoML#53]
Currently, I think AutoH2OModeler does not have the option for quantile regression. H2o has a quantile regression for GBM only so far and this option is not available for XGBoost. Do you have a plan to add a prediction interval for XGBoost with H2o? Or there is any way we can do it.

I tried to find lower and upper interval using the function from this (https://towardsdatascience.com/regression-prediction-intervals-with-xgboost-428e0a018b)
And attempted to add an interval to the prediction of XGBoost of H2o. However, by using the above-mentioned function for the quantile XGBoost interval are flat and the prediction is going beyond the range of the lower and upper interval. I will appreciate your suggestion.

**Comment by AdrianAntico** [comment]
@Shafi2016 I'm not currently doing anything with the AutoH2OModeler() function. I have quantile regression available with that function if you use h2o gbm or h2o DL. You can also use AutoH2oGBMRegression() and AutoCatBoostRegression() for quantile regression by setting the Alpha argumenty, although I believe h2o returns better results. I do not have any current plans to add in quantile regression for the xgboost versions.

**Comment by Shafi2016** [comment]
Thanks, Adrian for the prompt response. Let me check it with the AutoH2oGBMRegression(). If I can get what I am looking for.  I have the same conclusion that h2o XGBoost provides better accuracy and is much better than the other packages. I checked almost every package both in python and R.

**Comment by AdrianAntico** [comment]
@Shafi2016 I know with XGBoost I just need to set up a pinball loss function and then XGBoost can deliver quantile regression. I just haven't spent the time on that at this point.

**Comment by Shafi2016** [comment]
That will be a good addition. As currently, this is not available for XGBoost in any package of R and Python.
